pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -514,3 +517,145 @@ func ValidateMutatingWebhookConfigurationUpdate(newC, oldC *admissionregistratio
 		requireUniqueWebhookNames:               mutatingHasUniqueWebhookNames(oldC.Webhooks),
 	})
 }
+
+// ValidateValidatingAdmissionPolicy validates a ValidatingAdmissionPolicy before creation.
+func ValidateValidatingAdmissionPolicy(p *admissionregistration.ValidatingAdmissionPolicy) field.ErrorList {
+	return validateValidatingAdmissionPolicy(p)
+}
+
+func validateValidatingAdmissionPolicy(p *admissionregistration.ValidatingAdmissionPolicy) field.ErrorList {
+	allErrors := genericvalidation.ValidateObjectMeta(&p.ObjectMeta, false, genericvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrors = append(allErrors, validateValidatingAdmissionPolicySpec(&p.Spec, field.NewPath("spec"))...)
+
+	return allErrors
+}
+
+func validateValidatingAdmissionPolicySpec(spec *admissionregistration.ValidatingAdmissionPolicySpec, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	if spec.FailurePolicy != nil && !supportedFailurePolicies.Has(string(*spec.FailurePolicy)) {
+		allErrors = append(allErrors, field.NotSupported(fldPath.Child("failurePolicy"), *spec.FailurePolicy, supportedFailurePolicies.List()))
+	}
+	allErrors = append(allErrors, validateParamSource(&spec.ParamSource, fldPath.Child("paramSource"))...)
+	allErrors = append(allErrors, validateMatchResources(&spec.MatchConstraints, fldPath.Child("matchConstraints"))...)
+	if len(spec.Validations) == 0 {
+		allErrors = append(allErrors, field.Required(fldPath.Child("validations"), ""))
+	}
+	for i, validation := range spec.Validations {
+		allErrors = append(allErrors, validateValidation(&validation, fldPath.Child("validations").Index(i))...)
+	}
+	return allErrors
+}
+
+func validateParamSource(ps *admissionregistration.ParamSource, fldPath *field.Path) field.ErrorList {
```

Comment:
I was imagining we would only allow CRDs to be used as param sources when I originally wrote the KEP.  Enforce with a validation check?